### DISCUSSION TITLE
How to Update brunch

### DISCUSSION THREAD

**Discussion by patodia1110** [sebanc/brunch#1184] [General]
How can I update brunch version ?
Please anyone one tell me

**Comment by darethehair** [comment]
On the main page, there are instructions for either "Update both ChromeOS and the Brunch framework" or "Update only the Brunch framework".  I have typically downloaded the latest brunch 'xxx.tar.gz' file within ChromeOS, and run this command while in the 'shell'.  Alternatively, the main page also describes how to download/install/run "Brunch toolkit", which has an option to update Brunch as well.

My recent challenge is that some of the devices I own with only 16gb of storage no longer have room to download and install the latest Brunch, which is forcing me to upgrade (where I can) the storage to allow this to be done.

**Reply by gjlamb** [reply]
“ and run this command while in the 'shell' ”

What command?

**Reply by darethehair** [reply]
I probably should have said "run the required command" rather than "run this command", but in my case it was typically the one just to update the framework (since I was relying on the automatic update feature to handle ChromeOS itself):

sudo chromeos-update -f < path to the Brunch release archive >

As mentioned in the online docs, though, there is also the option to use the 'toolkit' to do this stuff through a menu.

In the meantime, yes, I did need to buy/replace the storage on two Chromeboxes in order to have more 'room' to do these upgrades.

**Comment by tejasraman** [comment]
The best way to update Brunch is to install the Brunch PWA. Edit your grub.cfg file, and add the “PWA” option to it, and then visit the page and click “Install”. Then you can update both brunch and ChromeOS from the app, as well as enable Brioche(brunch and Linux dual boot) and change over to unstable Brunch.

**Reply by nituladaboneko** [reply]
where do i go to read instructions on brunch pwa
thanks

**Reply by nituladaboneko** [reply]
ok found this https://github.com/sebanc/brunch-pwa
now looking for the cf framework options instructions on a readme page

**Reply by nituladaboneko** [reply]
ah buried in a tree branch in the readme https://github.com/sebanc/brunch/blob/master/troubleshooting-and-faqs.md

**Reply by nituladaboneko** [reply]
this looks like a easy way https://github.com/BiteDasher/brcr-update
here goes...

crosh> shell
chronos@localhost / $ curl -L -o - https://git.io/JLh1V | sudo bash

We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.


Done.
chronos@localhost / $ brcr-update
===> Downloading recovery.json file
################################################################################################################ 100.0%
===> Downloading brunch releases information
 -#O=#    #    #                                                                                                      
===> Parsing...
Board name:                rammus

Latest ChromeOS build:     14695.107.0
Current ChromeOS build:    13816.64.0
Latest ChromeOS version:   102
Current ChromeOS version:  90

Latest available brunch:   r102-stable-20220625
Current brunch:            r89-20210403

New version of ChromeOS available!
New version of brunch available!

Click Enter to continue

**Reply by tejasraman** [reply]
Use brcr. The process of enabling the pwa was different with the old config menu.

Once you have used brcr to update to r102, use this guide for settings updates: https://user-images.githubusercontent.com/96961804/163502788-cfe07d96-768c-4f71-be95-a15808a2b464.mp4

**Reply by nituladaboneko** [reply]
wow cool video thanks
should be on discord also

i just ran brcr and ran out of space. so the 16gb standard ssd on chrome device not enough to do the in place upgrade:


Click Enter to continue
> 
===> Downloading brunch r102-stable-20220625
###################################################################################### 100.0%
===> Downloading ChromeOS 14695.107.0
###################################################################################### 100.0%
===> Unpacking ChromeOS .zip archive...
Archive:  /usr/local/brcr_updater/chromeos.bin.zip
  inflating: chromeos_14695.107.0_rammus_recovery_stable-channel_mp-v2.bin  
chromeos_14695.107.0_rammus_recovery_stable-channel_mp-v2.bin:  write error (disk full?).  Continue? (y/n/^C) n

warning:  chromeos_14695.107.0_rammus_recovery_stable-channel_mp-v2.bin is probably truncated
chronos@localhost / $ 


prob same issue with pwa

gonna have to order, install larger ssd, reinstall mrchromebox, brunch, and then try pwa

what a pain!

**Comment by tejasraman** [comment]
Just use the pwa
(Edit - you need to update to use the pwa)

**Reply by nituladaboneko** [reply]
didn't know what a pwa was until now
thanks

**Comment by tejasraman** [comment]
God... what chromebook do you have?
I recently tried Brunch on a Samsung Chromebook 3 (Reached EOL this month) and the 16gb storage wasn't enough - and MrChromebox warns you against using the CELES (Samsung Chromebook 3)

Is it a C720?

**Comment by nituladaboneko** [comment]
i have many chrome devices but the one in question is a zako (c710)
i'm typing on it now.

**Comment by tejasraman** [comment]
I dunno, my CELES filled up after a few updates..

**Comment by tejasraman** [comment]
Also, Zako is the HP Chromebook G1. Yours is a PARROT

**Comment by nituladaboneko** [comment]
Thanks.

 

Zako is the HP ChromeBOX G1.

 

Yours is a PARROT C720 Chromebook.

 

I have both.

 

 
 

Sent: Thursday, July 07, 2022 at 5:30 AM

**Comment by tejasraman** [comment]
1. Try chrome os flex (Brunch takes up extra space on your eMMC)
2. You’re confusing me. Zako = chromebox, Parrot = c710, Peppy = c720, Celes (what I had) = Samsung Chromebook 3
